trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/TrinoAwsProxyServerModule.java

```diff
@@ -147,8 +151,13 @@ protected void setup(Binder binder)
             return AssumedRoleProvider.NOOP;
         });
 
+        // RemoteS3 binder
+        configBinder(binder).bindConfig(RemoteS3Config.class);
+
+        // RemoteS3 provided implementations
+        install(remoteS3Module(DEFAULT_REMOTE_S3_IDENTIFIER, DefaultRemoteS3Facade.class, new DefaultRemoteS3Module()));
```

Comment:
Yep I agree with Jordan. This might work since you set "default" as the default config for `remote-s3.type`, but it doesn't follow the pattern we have for other plugins.

Reply:
It's a bit different for `RemoteS3Facade` plugin as the default implementation also have some dependencies, whereas for other plugins the default is `NOOP`. Another way would be to conditionally bind `DefaultRemoteS3Module` module on `RemoteS3Config.getIdentifier().isEmpty()`.

Reply:
Changed to `conditionalModule(RemoteS3Config.class, ..)`